src/main/java/Task.java

```diff
@@ -0,0 +1,132 @@
+        }
+        String command = sTokens[0];
+        String taskString = sTokens[1];
+        switch (command) {
+            case "todo":
+                return new ToDo(taskString);
+            case "deadline":
+                // Further split the remaining string based on /by
+                if (!taskString.contains("/by")) {
+                    throw new CreateDeadlineException(
+                            "You can't expect me to track a deadline if you don't give me the.. deadline right? haha\n" +
+                                    "please include '/by' followed by the deadline :)");
+                }
+                String[] descriptionNBy = taskString.split("/by");
+                if (descriptionNBy.length > 2) {
+                    throw new CreateDeadlineException("Don't procrastinate with multiple deadlines !! :)");
+                }
+                return new Deadline(descriptionNBy[0].trim(),
+                                    descriptionNBy[1].trim());
+            case "event":
+                // Further split the remaining string based on /from and /to
+                if (!taskString.contains("/from") || !taskString.contains("/to")) {
+                    throw new CreateEventException(
+                            "I'm not sure how you're going for an event without a start time or end time\n" +
+                                    "please include '/from' and '/to' so I can keep track for you :)");
+                }
+                if (taskString.indexOf("/from") > taskString.indexOf("/to")) {
+                    throw new CreateEventException("Sorry! /from must appear before /to :)");
+                }
+                String[] descriptionNFromNTo = taskString.split("/from |/to ");
+
+                if (descriptionNFromNTo.length > 3) {
+                    throw new CreateEventException("um.. I'm not sure how I can track for you an event that has\n" +
+                            "more than 1 start or end time! :)");
+                }
+                return new Event(descriptionNFromNTo[0].trim(),
+                                 descriptionNFromNTo[1].trim(),
+                                 descriptionNFromNTo[2].trim());
+        }
+        return null; // shouldn't come here, since we have done the check already before calling this factory function
+    }
+
+    /**
+     * Returns a new Task created from the parsedString taken from the saved data file.
+     *
+     * @param parsedString string from data file
+     * @return a subclass of Task (ToDo, Event, Deadline)
+     * @throws CreateTaskException if save file corrupted or edited such that format is not what is expected
+     */
+    public static Task createFromParsedString(String parsedString) throws CreateTaskException {
+        // Get trimmed tokens
+        // ToDos -> T | y or n | description
+        // Deadlines -> D | y or n | description | by
+        // Events -> E | y or n | description | from | to
```

Comment:
This could be in your JavaDoc comment or maybe in a single multi-line comment instead, using /* ... */